File: client/src/types/course.ts
export interface Category {
  id: string;
  name: string;
  description?: string;
}

export interface Course {
  id: string;
  title: string;
  description: string;
  thumbnailUrl?: string;
  price: number;
  level: "Beginner" | "Intermediate" | "Advanced";
  status: "Draft" | "Published";
  instructorName: string;
  categoryName: string;
  instructorId: string;
  categoryId: string;
  enrollmentCount: number;
}

export interface CourseDetail extends Course {
  modules: CourseModule[];
}

export interface CourseModule {
  id: string;
  title: string;
  order: number;
  lessons: LessonSummary[];
}

export interface LessonSummary {
  id: string;
  title: string;
  order: number;
  durationMinutes: number;
  type: string;
}

export interface CreateCourseData {
  title: string;
  description: string;
  price: number;
  level: "Beginner" | "Intermediate" | "Advanced";
  categoryId: string;
  thumbnailUrl?: string;
}

export interface Certificate {
  id: string;
  studentId: string;
  enrollmentId: string;
  studentName: string;
  courseTitle: string;
  verificationCode: string;
  issuedAt: string;
}

File: server/src/TUTORA.Domain/Entities/Course.cs
using TUTORA.Domain.Common;
using TUTORA.Domain.Entities.Identity;
using TUTORA.Domain.Enums;

namespace TUTORA.Domain.Entities;

public class Course : BaseEntity
{
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string? ThumbnailUrl { get; set; }
    public decimal Price { get; set; }
    public CourseLevel Level { get; set; }
    public CourseStatus Status { get; set; } = CourseStatus.Draft;
    public Guid InstructorId { get; set; }
    public Guid CategoryId { get; set; }

    public ApplicationUser Instructor { get; set; } = null!;
    public CourseCategory Category { get; set; } = null!;
    public ICollection<Module> Modules { get; set; } = [];
    public ICollection<Enrollment> Enrollments { get; set; } = [];
}